services/core/java/com/android/server/policy/PhoneWindowManager.java

```diff
@@ -4463,6 +4464,13 @@ private boolean unpinActivity(boolean checkOnly) {
     /** {@inheritDoc} */
     @Override
     public KeyEvent dispatchUnhandledKey(WindowState win, KeyEvent event, int policyFlags) {
+        boolean isDesktopInteractive =
+                SystemProperties.getBoolean(PROPERTY_MARUOS_DESKTOP_INTERACTIVE, false);
+        InputDevice device = event == null ? null : event.getDevice();
+        boolean isExternal = device != null && device.isExternal();
+        if (isDesktopInteractive && isExternal) {
+            return null;
```

Comment:
Can we use a debug log statement to log that this key is being ignored when debug flag is on? May be useful if we are debugging something and this unintentionally ignores a key we want. Something like:

```java
if (DEBUG_INPUT) {
    Slog.d(TAG, "Ignoring key event while desktop is interactive");
}
```